hayeon/programmers-1/콜라츠-추측.js

```diff
@@ -0,0 +1,32 @@
+function solution(num) {
+  let result = 0;
+
+  for (let i = 0; i <= 500; i++) {
+    if (num !== 1) {
+      num = num % 2 === 0 ? num / 2 : num * 3 + 1;
+    } else {
+      return (result = i);
+    }
+  }
+  return (result = -1);
+}
```

Comment:
```suggestion
function solution(num) {
  for (let i = 0; i <= 500; i++) {
    if (num === 1) return i;
    num = num % 2 === 0 ? num / 2 : num * 3 + 1;
  }
  return -1;
}
```
- 잘푸셨습니다 👍 while문이 아닌 `for`문으로 접근하는 것도 좋아보이네요 :) while문이나 for문이나 결국 연산 횟수는 같겠지만 ㅎㅎㅎ
- 위에 코드는 return 해야줘야 되는 조건문을 위로 빼고, 불 필요한 변수를 제거했습니다 굳이 `result` 에다 i랑 -1을 할당하고 리턴 할 필요 없이 `바로 값을 return`해주는게 깔끔해보여요~ 🙆‍♂️
- 아래는 제 while문으로 접근한 코드입니다 참고해보세요 ~
```js
function solution(num) {
  let result = 0;

  while (num > 1) {
    if (result > 500) return -1;

    num = num % 2 === 0 ? num / 2 : num * 3 + 1;
    result++;
  }

  return result;
}
```

Reply:
헉 훨씬 깔끔하네요!!!!! 와우,,, 대박입니다,, while 문으로도 풀 수 있군여,,, 참고해보겠습니다!!!